### PR TITLE
doc/user: port ttl pattern guide

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

Port `lts` [TTLs guide](https://materialize.com/docs/lts/sql/patterns/ttl/)

### Tips for reviewer

Simple porting PR.

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: doc/user/content/sql/patterns/ttl.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+## Pattern
+
+The pattern uses a temporal filter over a row's creation timestamp plus a TTL. The sum represents the row's **expiration time**. After reaching the expiration time, the query will drop the row.
+
+Pattern example:
+```sql
+  CREATE VIEW TTL_VIEW
+  SELECT (created_ts + ttl) as expiration_time
+  FROM events
+  WHERE mz_logical_timestamp() < (created_ts + ttl);
```
Comment (ahelium) [inline doc/user/content/sql/patterns/ttl.md:29]: mz_logical_timestamp() -> mz_now() ?

Comment (joacoc) [reply doc/user/content/sql/patterns/ttl.md:29]: Wohoo. Nice catch!

Comment (ahelium) [reply doc/user/content/sql/patterns/ttl.md:29]: CREATE VIEW TTL_VIEW AS...

**Review by ahelium — CHANGES_REQUESTED** [review]
This is really neat! I think we need a clear reference to the mz_now() docs alongside the temporal filters docs.

**Comment by joacoc** [comment]
Thank you, @ahelium! Changes applied.

Now `mz_now()` appears as content to understand besides `temporal filters` at the beginning.

File: doc/user/content/sql/patterns/ttl.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+Before continuing, make sure to understand how [mz_now()](/sql/functions/now_and_mz_now/) and [temporal filters](/sql/patterns/temporal-filters/) work.
+
+## Pattern
+
+The pattern uses a temporal filter over a row's creation timestamp plus a TTL. The sum represents the row's **expiration time**. After reaching the expiration time, the query will drop the row.
+
+Pattern example:
+```sql
+  CREATE VIEW TTL_VIEW
```
Comment (ahelium) [inline doc/user/content/sql/patterns/ttl.md:26]: I think you are still missing an AS, here, but otherwise LGTM!

**Review by ahelium — APPROVED** [review]

File: doc/user/content/sql/patterns/ttl.md (hunk 2 above)
Comment (joacoc) [reply doc/user/content/sql/patterns/ttl.md:26]: Correct!

**Pull request merged**